src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/hooks/PatchFeatureFlagUtils.ts

```diff
@@ -0,0 +1,199 @@
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import { isEqual } from 'lodash-es'
+import { v4 as uuid } from 'uuid'
+import type { FeatureState, TargetMap } from 'services/cf'
+import patch from '@cf/utils/instructions'
+import type { FormVariationMap, TargetGroup, TargetingRulesFormValues } from '../Types.types'
+
+// Utils class to help encapsulate the complexity around patch instruction creation and hide this from the components
+interface PatchFeatureFlagUtilsReturn {
+  hasFlagStateChanged: () => boolean
+  hasDefaultOnVariationChanged: () => boolean
+  updateFlagState: () => void
+  updateDefaultServe: () => void
+  addedTargetGroups: (formVariation: FormVariationMap) => TargetGroup[]
+  removedTargetGroups: (formVariation: FormVariationMap) => TargetGroup[]
+  addedTargets: (formVariation: FormVariationMap) => string[]
+  removedTargets: (formVariation: FormVariationMap) => string[]
+  percentageRolloutAdded: () => boolean
+  percentageRolloutUpdated: () => boolean
+  percentageRolloutRemoved: () => boolean
+  createAddTargetGroupInstructions: (formVariation: FormVariationMap, targetGroups: TargetGroup[]) => void
+  createRemoveTargetGroupsInstructions: (targetGroups: TargetGroup[]) => void
+  createAddTargetsInstructions: (formVariation: FormVariationMap, targetIds: string[]) => void
+  createRemoveTargetsInstructions: (formVariation: FormVariationMap, targetIds: string[]) => void
+  createAddPercentageRolloutInstructions: () => void
+  createUpdatePercentageRolloutInstructions: () => void
+  createRemovePercentageRolloutInstructions: () => void
+}
+
+export const PatchFeatureFlagUtils = (
+  submittedValues: TargetingRulesFormValues,
+  initialValues: TargetingRulesFormValues
+): PatchFeatureFlagUtilsReturn => {
+  const hasFlagStateChanged = (): boolean => submittedValues.state !== initialValues.state
+
+  const hasDefaultOnVariationChanged = (): boolean => submittedValues.onVariation !== initialValues.onVariation
+
+  const createUpdateFlagStateInstruction = (): void =>
+    patch.feature.addInstruction(patch.creators.setFeatureFlagState(submittedValues.state as FeatureState))
+
+  const createDefaultServeInstruction = (): void =>
+    patch.feature.addInstruction(patch.creators.updateDefaultServeByVariation(submittedValues.onVariation))
+
+  const addedTargetGroups = (formVariation: FormVariationMap): TargetGroup[] => {
+    const intialTargetGroups: TargetGroup[] = formVariation.targetGroups
+
+    // get the submitted target groups for the given formVariation
+    const submittedTargetGroups: TargetGroup[] = submittedValues.formVariationMap.filter(
+      variation => variation.variationIdentifier === formVariation.variationIdentifier
+    )[0].targetGroups
+
+    return submittedTargetGroups.filter(
+      submittedTargetGroup =>
+        !intialTargetGroups
+          .map(intialTargetGroup => intialTargetGroup.identifier)
```

Comment:
SUPER pedantic, so please feel free to ignore, but this would be a great place for destructuring
```suggestion
          .map(({ identifier }) => identifier)
```

Reply:
fixed